src/server/api/index.js

```diff
@@ -133,4 +135,19 @@ router.get('/addanalytics/:type', (req, res) => {
   }
 });
 
+router.get('/analyticsfile', (req, res) => {
+  try {
+    if (!CheckRoleServices.isAdmin(res.locals.token.roles)) {
+      return res.status(403).send(new Error('Unauthorized role'));
+    }
+    Analytics.getAnalytics().then(result => {
+      const excel = json2xls(result);
+      fs.writeFileSync('analytics.xlsx', excel, 'binary');
+      res.download('analytics.xlsx');

```

Comment:
Is there a way to not have the downloaded file added to the repo? either by deleting or by putting it in a folder, then adding the folder to .gitignore maybe

Reply:
Right now when I download the file, a copy is created in our repo and it can get messy if you do it multiple times

Reply:
Also for that one linting error I think you can temporarily resolve it by capitalizing the comment

Reply:
@finnzink You will also want to make sure the file you create has an unique filename when creating it on the local file system. See https://www.npmjs.com/package/tmp as a package that can help.

You might run into race conditions if two people ask for an excel download at the same time and it gets deleted while still trying to be sent. Or overwritten by the other request. 

Create a tmp folder under src/server. You will need a .gitkeep file to commit an empty directory to git. See https://sailsjs.com/documentation/anatomy/api/policies/.gitkeep

Add other files added to this tmp folder in .gitignore.

Reply:
@rlorenzo ok so if I make a new file (w/ unique name) for every request, how do I get rid of the old files? Should I just delete them? In which case wouldn't that cause a similar problem (could get deleted while someone else is downloading)

Reply:
@finnzink  yes cleanup is essential. The docs say “ You can set whether you want to remove the temporary file on process exit or not.”

Reply:
@rlorenzo Got it. Thanks

Reply:
@finnzink I don't think it would be fighting it. The docs clearly say:

> If you do not want to store your temporary directories and files in the standard OS temporary directory, then you are free to override that as well.

But since files are being created in a temp folder outside the code directory, then that's great. No need to change anything.